In [2]:
import pandas as pd
ds1 = pd.read_csv('general_data.csv')
ds1.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [6]:
ds1.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

# HYPOTHESIS 1:

In [4]:
from sklearn import preprocessing
label_encoder= preprocessing.LabelBinarizer()
ds1['Attrition']= label_encoder.fit_transform(ds1['Attrition'])
from scipy.stats import pearsonr
r,p = pearsonr(ds1.Attrition,ds1.Age)
print(r,p)

-0.15920500686577965 1.996801615886744e-26


p-value= 1.996801615886744e-26

Since p-value < 0.05 , Ho(Null Hypothesis) is rejected.

H0: There is no significant correlation between age and attrition of employees.

Ha: There is significant correlation between age and attrition of employees.

# HYPOTHESIS 2:

In [5]:
r,p = pearsonr(ds1.Attrition,ds1.DistanceFromHome)
print(r,p)

-0.009730141010179674 0.5182860428050771


p-value=0.518

Since p-value > 0.05 , H0 is accepted.

H0: There is no significant correlation between distance from home and attrition of employees.

Ha: There is significant correlation between distance from home and attrition of employees.

# HYPOTHESIS 3:

In [7]:
r,p = pearsonr(ds1.Attrition,ds1.MonthlyIncome)
print(r,p)

-0.031176281698115007 0.03842748490600132


p-value=0.038

Since p-value < 0.05, H0 is rejected.

H0: There is no significant correlation between Monthly income and attrition of employees.

Ha: There is significant correlation between Monthly income and attrition of employees.

# HYPOTHESIS 4:

In [8]:
r,p = pearsonr(ds1.Attrition,ds1.PercentSalaryHike)
print(r,p)

0.03253259489105349 0.030743386433355353


p-value=0.0307

Since p-value < 0.05, H0 is rejected.

H0: There is no significant correlation between Percent salary hike and attrition of employees.

Ha: There is significant correlation between Percent salary hike and attrition of employees.

# HYPOTHESIS 5:

In [9]:
r,p = pearsonr(ds1.Attrition,ds1.TrainingTimesLastYear)
print(r,p)

-0.049430576244255 0.0010247061915365072


p-value=0.00102

Since p-value < 0.05, H0 is rejected.

H0: There is no significant correlation between Training time last year and attrition of employees.

Ha: There is significant correlation between Training time last year and attrition of employees.

# HYPOTHESIS 6:

In [11]:
r,p = pearsonr(ds1.Attrition,ds1.YearsAtCompany)
print(r,p)

-0.1343922139899772 3.1638831224877484e-19


p-value=3.1638831224877484e-19

Since p-value < 0.05, H0 is rejected.

H0: There is no significant correlation between Years at company and attrition of employees.

Ha: There is significant correlation between from Years at company and attrition of employees.

# HYPOTHESIS 7:

In [12]:
r,p = pearsonr(ds1.Attrition,ds1.YearsSinceLastPromotion)
print(r,p)

-0.03301877514258434 0.028330336189396753


p-value=0.028

Since p-value < 0.05, H0 is rejected.

H0: There is no significant correlation between Years since last promotion and attrition of employees.

Ha: There is significant correlation between Years since last promotion and attrition of employees.

# HYPOTHESIS 8:

In [13]:
r,p = pearsonr(ds1.Attrition,ds1.YearsWithCurrManager)
print(r,p)

-0.15619931590162847 1.7339322652896276e-25


p-value=1.733

Since p-value > 0.05, H0 is accepted.

H0: There is no significant correlation between Years with current manager and attrition of employees.

Ha: There is significant correlation between Years with current manager and attrition of employees.